### ISSUE TITLE
Consistent .train() and .eval() calls?

### ISSUE THREAD

**Issue by iamgroot42** [MinChen00/Face-Auditor#1]
More of a question than an 'issue':
https://github.com/MinChen00/Face-Auditor/blob/f9942d2958fee769077d758534fb4d6969c5944f/lib_model/relation_net/relation_net.py#L66
For the above (and subsequent calls to `self.model.train()`), should `self.feat_encoder` also not be set to `train()`?
The same question for when `self.model` is set to `eval()`:
https://github.com/MinChen00/Face-Auditor/blob/f9942d2958fee769077d758534fb4d6969c5944f/lib_model/relation_net/relation_net.py#L124

This should not lead to any issue while training (since gradients are cleared out anyway) but may affect performance if layers like `Dropout` and `BatchNorm` are present.

**Comment by MinChen00** [comment]
Hi, I checked the original implementation of the relationnet (I referred to [this link](https://github.com/floodsung/LearningToCompare_FSL.git).) It indeed needs to initialize the parameters of both the relation model and the feat_encoder before training (see the `weights_init` function.) And Line 96-97 also manually initialize the weights. So the current version should not cause any issues to train a usable relation model.

**Comment by iamgroot42** [comment]
Yes, the ability to train is not an issue. The problem is- when evaluating the model, the feature encoder is not set to `eval()` mode which means, it would continue to apply dropout (if present) and update batch-norm parameters, which is undesirable

**Comment by iamgroot42** [comment]
I looked at the implementation you referred to, and seems like the original codebase also has this issue (https://github.com/floodsung/LearningToCompare_FSL/issues/12) - fixing it in their case dropped test accuracy of models by as much as 10%, and it would definitely have the potential to change results in this setting as well
